pkg/registries/registries.go

```diff
@@ -159,7 +159,7 @@ func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureScop
 	}
 	for _, mirrorSet := range mirrorSets {
 		reg := getRegistryEntry(mirrorSet.Source)
-		reg.MirrorByDigestOnly = true
+		reg.MirrorByDigestOnly = mirrorSet.AllowMirrorByTags
```

Comment:
Shouldn't this be a "not"?
```suggestion
		reg.MirrorByDigestOnly = !mirrorSet.AllowMirrorByTags
```